src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -125,64 +164,59 @@ def __init__(
     @cached_property
     def _mounts_snapshot(self) -> list['Mount']:
         """Returns all mounts, sorted by longest prefixes first."""
-        return sorted(self._mounts_crawler.snapshot(), key=lambda _: len(_.name), reverse=True)
+        return sorted(self._mounts_crawler.snapshot(), key=lambda _: (len(_.name), _.name), reverse=True)
 
     def _external_locations(self) -> Iterable[ExternalLocation]:
-        min_slash = 2
-        external_locations: list[ExternalLocation] = []
+        trie = LocationTrie()
         for table in self._tables_crawler.snapshot():
-            location = table.location
-            if not location:
+            table = self._resolve_location(table)
+            if not table.location:
                 continue
-            # TODO: refactor this with LocationTrie
-            if location.startswith("dbfs:/mnt"):
-                location = self.resolve_mount(location)
-            if not location:
+            trie.insert(table)
+        queue = list(trie.children.values())
+        external_locations = []
+        while queue:
+            curr = queue.pop()
+            num_children = len(curr.children)  # 0 - take parent
+            if curr.location and (num_children > 1 or num_children == 0):
```

Comment:
Isn't the following the same?
```suggestion
            if num_children != 1:
```

Reply:
Also explain why we can simply extend when there is one child